src/garage/envs/environment.py

```diff
@@ -0,0 +1,189 @@
+
+        If the environment returned a `TimeStep` with `StepType.LAST` at the
+        previous step, this call to `step` will start a new sequence and
+        `action` will be ignored.
+
+        This method will also start a new sequence if called after the
+        environment has been constructed and `reset` has not been called.
+        Again, in this case `action` will be ignored.
+
+        Args:
+            action (np.ndarray): A NumPy array, or a nested dict, list or tuple
+                of arrays corresponding to `action_spec()`.
+
+        Returns:
+            garage.TimeStep: The TimeStep after the action. Attributes:
+                * env_spec (garage.envs.EnvSpec): Environment specification.
+                * observation: an `np.ndarray`, or a nested dict, list or
+                    tuple corresponding to :obj:`env_spec.observation_space`.
+                    The observation before applying the action.
+                * action (np.ndarray): A NumPy array, or a nested dict, list or
+                    tuple of arrays corresponding to `action_spec()`.
+                * reward (float): The reward for taking the action given the
+                    observation, at the this time step.
+                * next_observation: an `np.ndarray`, or a nested dict, list or
+                    tuple corresponding to :obj:`env_spec.observation_space`.
+                    The observation after applying the action.
+                * terminal (bool): The termination signal for the this time
+                    step.
+                * env_info (dict): A dict arbitrary environment state
+                    information.
+                * agent_info (numpy.ndarray): A dict of arbitrary agent
+                    state information.
+                * step_type (garage.StepType): a `StepType` enum value. Can
+                    either be StepType.FIRST, StepType.MID, StepType.LAST,
+                    or StepType.TIMEOUT.
+
+        """
+
+    @abc.abstractmethod
+    def render(self, mode):
+        """Renders the environment.
+
+        The set of supported modes varies per environment. By convention,
+        if mode is:
+
+        * rgb_array: Return an numpy.ndarray with shape (x, y, 3),
+          representing RGB values for an x-by-y pixel image, suitable
+          for turning into a video.
+        * ansi: Return a string (str) or StringIO.StringIO containing a
+          terminal-style text representation. The text can include newlines
+          and ANSI escape sequences (e.g. for colors).
+
+        Make sure that your class's metadata 'render.modes' key includes
+        the list of supported modes.
+
+        Args:
+            mode (str): the mode to render with. The string must be present in
+            self.metadata['render.modes'].
+
+        Raises:
+            NotImplementedError: If the mode is not implemented.
+
+        For example:
+
+        class MyEnv(Env):
```

Comment:
make sure this renders nicely on ReadTheDocs

Reply:
I can't get this to render nicely on ReadTheDocs. I tried https://recommonmark.readthedocs.io/en/latest/auto_structify.html#codeblock-extensions but it is not working. Any examples?